src/traces/scatter/attributes.js

```diff
@@ -67,7 +67,11 @@ module.exports = {
     },
     ids: {
         valType: 'data_array',
-        description: 'A list of keys for object constancy of data points during animation'
+        description: [
```

Comment:
> is there a way to automatically cast these to strings

Not at the moment. Casting within `data_array` fields is done during the calc step at the moment.

As `groups` will only be used in filter transforms at first, we could simplify cast `groups` items in the filter `calcTransform` handler.

Alternatively, we could add a block to `Plots.doCalcdata`, casting `groups` items once and for all. In order to behave correctly during `restyle` and `extendTraces`, this would (unfortunately) require adding `groups` to the recalc-attribute list [here](https://github.com/plotly/plotly.js/blob/95c56e840aecf0c0a8a99a71341895f6bacbe4ab/src/plot_api/plot_api.js#L1247).

I have no preference. Your call.